src/test/ui/impl-trait/generic-with-implicit-hrtb-without-dyn.rs

```diff
@@ -0,0 +1,9 @@
+// edition:2021
+
+fn ice() -> impl AsRef<Fn(&())> {
```

Comment:
It also ICEs on edition 2015.  Could you add a `check-pass` test on edition 2015?